src/services/group.service.ts

```diff
@@ -0,0 +1,86 @@
+import { sequelize } from "../data-access/connectDB";
+import { IGroup } from "../models";
+import { Group } from "../models";
+
+export class GroupService {
+  async getGroups(): Promise<IGroup[]> {
+    return await (await Group.findAll()).map((group) => group.toDomain());
+  }
+
+  async getGroupByID(groupId: string): Promise<IGroup> {
+    const group = await Group.findOne({
+      where: { id: groupId },
+    });
+
+    if (!group) {
+      throw `Group with id=${groupId} is not found`;
+    }
+
+    return group.toDomain();
+  }
+
+  async getGroupByName(groupName: string): Promise<IGroup> {
+    const group = await Group.findOne({
+      where: { name: groupName },
+    });
+
+    if (!group) {
```

Comment:
I wouldn't use errors for handling empty results

Reply:
Changed. Thanks.